wrapper/src/main/java/software/amazon/jdbc/plugin/cache/DataRemoteCachePlugin.java

```diff
@@ -140,33 +147,76 @@ private ResultSet cacheResultSet(String queryStr, ResultSet rs, int expiry) thro
-      // Handle "no cache".
-      if (!StringUtils.isNullOrEmpty(tokens[0]) && "no".equals(tokens[0])) return null;
-      // Handle "cacheTTL=Xs"
-      if (!StringUtils.isNullOrEmpty(tokens[1]) && tokens[1].startsWith("ttl=")) {
-        int endIndex = tokens[1].indexOf('s');
-        if (endIndex > 0) {
+    LOGGER.finest("queryHint is: " + queryHint);
+    // Find CACHE_PARAM anywhere in the hint string (case insensitive)
+    String upperHint = queryHint.toUpperCase();
+    int cacheParamStart = upperHint.indexOf("CACHE_PARAM(");
+    if (cacheParamStart == -1) return null;
+
+    // Find the matching closing parenthesis
+    int paramsStart = cacheParamStart + 12; // Length of "CACHE_PARAM("
+    int paramsEnd = queryHint.indexOf(")", paramsStart);
+    if (paramsEnd == -1) return null;
+
+    // Extract parameters between parentheses
+    String cacheParams = queryHint.substring(paramsStart, paramsEnd).trim();
+    LOGGER.finest("Cache params: " + cacheParams);
+
+    // Empty parameters
+    if (StringUtils.isNullOrEmpty(cacheParams)) {
+      LOGGER.warning("Empty CACHE_PARAM parameters");
+      incrCounter(malformedHintCounter);
+      return null;
+    }
+
+    // Parse comma-separated parameters
+    String[] params = cacheParams.split(",");
+    Integer ttlValue = null;
+
+    for (String param : params) {
+      String[] keyValue = param.trim().split("=");
+      if (keyValue.length != 2) {
+        LOGGER.warning("Invalid parameter format: " + param);
+        incrCounter(malformedHintCounter);
+        return null;
+      }
+      String key = keyValue[0].trim().toLowerCase();
+      String value = keyValue[1].trim();
+
+      if ("ttl".equals(key)) {
+        // Parse TTL value (e.g., "300s")
+        if (value.endsWith("s")) {
           try {
-            return Integer.parseInt(tokens[1].substring(4, endIndex));
-          } catch (Exception e) {
-            LOGGER.warning("Encountered exception when parsing Cache TTL: " + e.getMessage());
+            int ttl = Integer.parseInt(value.substring(0, value.length() - 1));
+            // treat negative and 0 ttls as not cacheable
+            if (ttl <= 0) {
+              LOGGER.warning("TTL must be positive: " + ttl);
+              incrCounter(malformedHintCounter);
+              return null;
+            }
+            ttlValue = ttl;
+          } catch (NumberFormatException e) {
+            LOGGER.warning(String.format("Invalid TTL format of %s for query %s", value, queryHint));
+            incrCounter(malformedHintCounter);
+            return null;
           }
+        } else {
+          LOGGER.warning("TTL must end with 's': " + value);
+          incrCounter(malformedHintCounter);
+          return null;
         }
       }
+      // Handle other parameters in the future (otherParam=a, etc.)
     }
-
-    LOGGER.finest("Query hint " + queryHint + " indicates the query is not cacheable");
-    return null;
+    LOGGER.finest("ttl value is: " + ttlValue);
```

Comment:
No need to log this. I think the caller already logs the query is cacheable with some TTL?